nodes/src/nodes/llm_base/resilience.py

```
@@ -0,0 +1,439 @@
+
+class CircuitBreaker:
+    """Thread-safe circuit breaker with CLOSED / OPEN / HALF_OPEN states.
+
+    Parameters:
+        failure_threshold   -- Number of consecutive failures before opening (default 5).
+        recovery_timeout    -- Seconds to wait in OPEN before transitioning to HALF_OPEN (default 60).
+        half_open_max_calls -- Max trial calls allowed while HALF_OPEN (default 1).
+        provider_name       -- Human-readable label used in log messages and errors.
+    """
+
+    def __init__(
```

Ruff (0.15.7)

[warning] Missing return type annotation for special method `__init__`

Add return type annotation: `None`

(ANN204)